pkg/apis/core/validation/utils.go

```diff
@@ -195,3 +195,16 @@ func ValidateIPFamilies(ipFamilies []core.IPFamily, fldPath *field.Path) field.E
 
 	return allErrs
 }
+
+func validateDependencyWatchdogSettings(settings *core.SeedSettingDependencyWatchdog, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if settings.Weeder != nil && settings.Endpoint != nil && settings.Weeder.Enabled != settings.Endpoint.Enabled {
+		allErrs = append(allErrs, field.Forbidden(fldPath, "weeder and endpoint cannot have different values"))
+	}
+
+	if settings.Prober != nil && settings.Probe != nil && settings.Prober.Enabled != settings.Probe.Enabled {
+		allErrs = append(allErrs, field.Forbidden(fldPath, "prober and probe cannot have different values"))
+	}
+
+	return allErrs
+}
```

Comment:
This can be dropped as soon as you made sure that `gardener-apiserver` always keeps both values in sync.

Reply:
I am unsure about what to do in situation like below
`endpoint was set to true, then weeder was set to false in the next update` . We would need to prioritize either `endpoint` or `weeder` field in such case. 
For now I have not prioritized, but kept this validation intact , so that operator can bring both fields to the value he desires.
Pls let me know your suggestions.

Reply:
I don't understand, sorry. Why would we need to prioritise these fields at all if the `gardener-apiserver` always sets `endpoint=weeder`. How can they ever have different values?

Reply:
By priority I meant should I do `endpoint=weeder` or `weeder=endpoint` when both are set, but looks like you want the former. Will rebase and include the change